src/Text/Libyaml.hs

```diff
@@ -520,8 +553,13 @@ newtype ToEventRawException = ToEventRawException CInt
 instance Exception ToEventRawException
 
 decode :: MonadResource m => B.ByteString -> ConduitM i Event m ()
-decode bs | B8.null bs = return ()
-decode bs =
+decode = mapOutput yamlEvent . decodeMarked
+
+-- |
```

Comment:
Please add some comment here and to `decodeFileMarked` explaining the difference with `decode` and `decodeFile`.